src/pretalx/orga/views/speaker.py

```diff
@@ -265,10 +270,8 @@ def dispatch(self, request, event, code):
             person=self.request.user,
             orga=True,
         )
-        if request.GET.get("from") == "list":
-            return redirect(
-                reverse("orga:speakers.list", kwargs={"event": self.kwargs["event"]})
-            )
+        if referer := request.META.get("HTTP_REFERER"):
+            return redirect(referer)
```

Comment:
I would prefer to allow an explicit set of URLs here rather than redirecting to any header sent. Either the `from=` (with values list/profile/`submission:code`) or the standard `next=` (with the same checks for safe urls as in the login view) would be fine.